src/irmin/mem/irmin_mem.ml

```diff
@@ -39,8 +39,9 @@ module Read_only (K : Irmin.Type.S) (V : Irmin.Type.S) = struct
     t.t <- KMap.empty;
     Lwt.return_unit
 
-  let close _ =
+  let close t =
     [%log.debug "close"];
+    t.t <- KMap.empty;
```

Comment:
Are you sure this is the semantics we want here? Maybe that's fine, but this doesn't seem super trivial :-)

Reply:
I'm not sure at all :) (I was writing this comment https://github.com/mirage/irmin/pull/1794#issuecomment-1139714943 that discusses this choice)